src/hermes/model/path.py

```diff
@@ -0,0 +1,372 @@
+                _log.debug("Using type %s from %s.", item, _path)
+                return item
+
+            except (KeyError, IndexError, TypeError) as e:
+                _log.debug("%s: %s", _path, e)
+                _path = _path.parent
+                continue
+
+        return None
+
+    def _find_setter(self, target: dict | list, path: 'ContextPath', value: t.Any = None, **kwargs) -> t.Callable:
+        filter = {
+            'name': path._item,
+        }
+
+        if isinstance(path._item, str) or path._parent is not None:
+            filter['path'] = str(path)
+
+        if type := self._find_in_parent(target, path['@type']):
+            filter['type'] = type
+        elif value is not None:
+            match value:
+                case list(): filter['type'] = 'list'
+                case dict(): filter['type'] = 'map'
+        elif path._type is list:
+                filter['type'] = 'list'
+        elif path._type is dict:
+                filter['type'] = 'map'
+
+        if ep := kwargs.get('ep', None):
+            filter['ep'] = ep
+
+        setter = self.merge_strategies.select(**filter)
+        if setter is None:
+            return self._set_item
+        else:
+            return setter
+
+    def _set_item(self, target: dict | list, path: 'ContextPath', value: t.Any, **kwargs) -> t.Optional['ContextPath']:
+        match target, path._item:
+            case list(), int() as index if index < len(target):
+                match target[index]:
+                    case dict() as t: t.update(value)
+                    case list() as l: l[:] = value
+                    case _: target[index] = value
+
+            case dict(), str() as key if key in target:
+                match target[key]:
+                    case dict() as t: t.update(value)
+                    case list() as l: l[:] = value
+                    case _: target[key] = value
+
+            case dict(), str() as key:
+                target[key] = value
+            case list(), '*':
+                path._item = len(target)
+                target.append(value)
+            case list(), int() as index if index == len(target):
+                target.append(value)
+
+            case dict(), _ as key:
+                raise TypeError(f'Invalid key type {type(key)} to set in {path.parent}.')
+            case list(), int() as index:
+                raise IndexError(f'Index {index} out of bounds to set in {path.parent}.')
+            case list(), _ as index:
+                raise TypeError(f'Invalid index type {type(index)} to set in {path.parent}.')
+
+            case _, _:
+                raise TypeError(f'Cannot handle target type {type(target)} to set {path}.')
+
+        return value
+
+    def resolve(self, target: list | dict, create: bool = False, query: t.Any = None) -> ('ContextPath', list | dict, 'ContextPath'):
+        """
+        Resolve a given path releative to a given target.
```

Comment:
```suggestion
        Resolve a given path relative to a given target.
```

**typo:** THanks for fixing.